lib/spack/spack/test/environment_modifications.py

```diff
@@ -397,6 +397,12 @@ def test_sanitize_literals(env, blacklist, whitelist):
     ({'A_modquar': '1', 'b_modquar': '2', 'C_modshare': '3'},
      [r'(\w*)_mod(quar|share)'], [], [],
      ['A_modquar', 'b_modquar', 'C_modshare']),
+    # Check regex to blacklist Modules v5 related vars
+    ({'__MODULES_LMALTNAME': '1', '__MODULES_LMCONFLICT': '2'},
+     ['__MODULES_(.*)'], [], [], ['__MODULES_LMALTNAME', '__MODULES_LMCONFLICT']),
```

Comment:
I don't get the sense of this new test. Should we always blacklist by default these variables? Here we are just trying the same `sanitize` method with different inputs in unit-tests - so I don't really see a difference among this new test and the ones before it.

Reply:
The intent was not to add a new test. I do not know how this should be done but if you want to `sanitize` the user environment by clearing the `MODULES_LMALTNAME` environment variable, starting Modules 5.0 you should clear the `__MODULES_LMALTNAME` environment variable instead.